### PR TITLE
Add CLI argument to change files easily

### PR DESCRIPTION
**Pull request by robinschmid**
Added arguments i: input to change the input files in batch files easily. It just replaces all the files in the AllSpectralDataImportModule.

Either provide a txt file with paths or use glob patterns
```
-b D:\Data\batch\test.xml -i D:\Data\input_files.txt
-b D:\Data\batch\test.xml -i D:\Data\*.mzML
```

### CODE REVIEW

File: src/main/java/io/github/mzmine/main/MZmineArgumentParser.java [hunk 1]
```diff
@@ -102,6 +110,20 @@ public void parse(String[] args) {
         logger.info(() -> "Batch file set by command line: " + sbatch);
         batchFile = new File(sbatch);
       }
+
+      String sinput = cmd.getOptionValue(input.getLongOpt());
+      if (sinput != null) {
+        logger.info(() -> "Input files were set to: " + sinput);
+        // search for files
+        try {
+          overrideDataFiles = FileAndPathUtil.parseFileInputArgument(sinput);
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/main/MZmineArgumentParser.java:119]: maybe only do this if it's headless mode?

File: src/main/java/io/github/mzmine/main/MZmineArgumentParser.java [hunk 2]
```diff
@@ -190,5 +212,9 @@ public boolean isLoadTdfPseudoProfile() {
   public boolean isLoadTsfProfile() {
     return loadTsfProfile;
   }
+
+  public File[] getOverrideDataFiles() {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/main/MZmineArgumentParser.java:216]: Nullable

File: src/main/java/io/github/mzmine/modules/batchmode/BatchModeModule.java [hunk 3]
```diff
@@ -88,6 +102,16 @@ public static ExitCode runBatch(@NotNull MZmineProject project, File batchFile,
         }
       }
 
+      if (overrideDataFiles != null) {
+        if (!newQueue.setImportFiles(overrideDataFiles)) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/batchmode/BatchModeModule.java:106]: the method never returns false but throws an exception instead, so the code below is never reached.

File: src/main/java/io/github/mzmine/modules/batchmode/BatchQueue.java [hunk 4]
```diff
@@ -225,4 +229,27 @@ public void saveToXml(final Element xmlElement) {
...
     }
   }
+
+  /**
+   * Replace all import files in the {@link AllSpectralDataImportModule} - which needs to be the
+   * first step in batch
+   *
+   * @param allFiles replaces import files
+   */
+  public boolean setImportFiles(final File[] allFiles) {
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/batchmode/BatchQueue.java:239]: does not return false but throws an exception. Refactor to do one of the two.

**Review by SteffenHeu — CHANGES_REQUESTED** [review]
looks good, but some comments

File: src/main/java/io/github/mzmine/main/MZmineArgumentParser.java [hunk 5]
```diff
@@ -62,6 +65,11 @@ public void parse(String[] args) {
     batch.setRequired(false);
     options.addOption(batch);
 
+    Option input = new Option("i", "input", true,
+        "input files. Either defined in a .txt text file with one file per line or ");
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/main/MZmineArgumentParser.java:69]: Changed to
```
 Option input = new Option("i", "input", true,
        """
    input data files. Either defined in a .txt text file with one file per line
    or by glob pattern matching. To match all .mzML files in a path: -i "D:\\Data\\*.mzML"
    """);
```

File: src/main/java/io/github/mzmine/modules/batchmode/BatchQueue.java (hunk 4 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/batchmode/BatchQueue.java:239]: returns false now. Logging was improved and jdocs as well

File: src/main/java/io/github/mzmine/modules/batchmode/BatchModeModule.java (hunk 3 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/batchmode/BatchModeModule.java:106]: returns false now. Logging was improved and jdocs as well

**Review by SteffenHeu — APPROVED** [review]

**Comment by SteffenHeu** [comment]
looks good, thanks

**Pull request merged**